### PR TITLE
cleaning up in preparation for a release version coming soon

### PR DESCRIPTION
**Pull request by rananna**

### CODE REVIEW

**Comment by stsdc** [comment]
Why not to put this things into wiki? :thinking:

File: README.md [hunk 1]
```diff
@@ -1,70 +1,32 @@
...
-* VLCD5 handle bar remote keypad has 4 buttons and 5 wires (1 wire for each button and 1 common wire to all buttons). Connect the wires directly to the wireless board.
-* For flashing the firmware, that you will need to do only one time, solder the 4 wires to the STLinkV2 (I use a cheap 1€ STLinkV2 clone that I bought on EBay).
-
-## Firmware
-Flash the firmware with any tool that works for you. The firmware HEX file to flash is this one: [ebike_wireless_remote_with_sd-v.0.5.hex](firmware/release/ebike_wireless_remote_with_sd-v0.5.hex)
-
+- Forum message: [Fully wireless ANT+ BLE TSDZ2 mid drive motor (like Specialized Turbo Levo) -- Flexible OpenSource firmware](https://endless-sphere.com/forums/viewtopic.php?t=106346)
+- Developers: see [documentation](documentation/README.md)
+- We are also developing a wireless ebike controller for the TSDZ2 motor: https://github.com/OpenSource-EBike-firmware/TSDZ2_wireless
+# Overview
```
Comment (stsdc) [inline README.md]: There shouldn't be more than one first level headers.

File: README.md [hunk 2]
```diff
@@ -1,70 +1,32 @@
...
+[Battery considerations](documentation/battery.md)
+-----
+-----
+[Further Development and Planned features](documentation/planned.md)
+------
+------
+[Getting Started](documentation/getting_started.md)
+-------
+-------
+##Questions/Comments?
```
Comment (stsdc) [inline README.md]: Missing space.

**Comment by casainho** [comment]
> Why not to put this things into wiki?

I only see disadvantages on using the wiki, as it will be another repository.
MD files are nice because on Code Studio, you can on left side edit and see the source file and on right side view a preview of the rendered page - this is an add on but it is great for developers.

**Comment by stsdc** [comment]
Why do You need another repository for Wiki?

When You edit wiki page you have Preview tab.

**Comment by casainho** [comment]
I think the wiki is another repository, if you want to  checkout to your local PC.

For you, what are the advantages of the wiki?

**Comment by stsdc** [comment]
Every repo has it's wiki. And it's separate version control. So no *.md files in the repo with a source code.

Any GitHub wiki can be cloned by appending wiki.git to the repo url, so the clone url for the repo `https://myorg/myrepo/` is: `git@github.com:myorg/myrepo.wiki.git` (for ssh) or `https://github.com/my/myrepo.wiki.git` (for https).

It will allow us to keep bare minimum in the readme and more specific things on the Wiki. It will be cleaner and easier to maintain.

**Comment by casainho** [comment]
> It will allow us to keep bare minimum in the readme and more specific things on the Wiki. It will be cleaner and easier to maintain.

Ok, I see what you mean. There is the need of a wiki that aggregates all the information, because the current 3 different repositories are about the EBike wireless (TSDZ2 only for now).

But there is already a wiki, of TSDZ2 firmware to where I planned to put the wireless, once we have a public release: https://github.com/OpenSource-EBike-firmware/TSDZ2_wiki/wiki

For that wiki, the image files or other files need to be on the https://github.com/OpenSource-EBike-firmware/TSDZ2_wiki while the wiki itself is on the https://github.com/OpenSource-EBike-firmware/TSDZ2_wiki.wiki.git
So this means anyone that want to edit the wiki and add / change a file, will need to make 2 different PRs??

I want to make the documentation / wiki / MD files, not available to anyone edit but instead always do PR for submit changes, that way I want to have a good control on the information because now with the wiki, I don´t have a easy way to know who edited the wiki, what and I don´t have notification...!! The wiki on github is a bit limited comparing to other wikis.

I would like to keep PRs for documentation and easy and fast way for users doing PRs, not the need to do 2 different PRs for changing the documentation.

**Comment by stsdc** [comment]
Every repo has its own wiki. You don't have to create separate repo.  I prefer to edit wiki online because I can see a final result, however nothing stops you from cloning wiki and [editing it on your machine](https://gist.github.com/subfuzion/0d3f19c4f780a7d75ba2).

If You don't want random people edit it, You can restrict acces to collaborators only. There is an option in Repo settings.
obraz
But, yes, Githuib Wiki has the downside: it is impossible to create PR to the Wiki. But [there are workarounds](https://stackoverflow.com/questions/10642928/how-to-pull-request-a-wiki-page-on-github), but they are not optimal.

[OpenSource-EBike-firmware/TSDZ2_wiki](https://github.com/OpenSource-EBike-firmware/TSDZ2_wiki/wiki) is very hard to navigate. Should be rewritten.

I see four options:

1. We choose [TSDZ2-Smart-EBike ](https://github.com/OpenSource-EBike-firmware/TSDZ2-Smart-EBike) as main repo for wiki
2. We use TSDZ2_wiki and synchronize it with https://github.com/OpenSource-EBike-firmware/TSDZ2_wiki.wiki.git (done by CI) (allows PRs)
2. We use another centralized medium, like https://www.gitbook.com/ 
3. We use multiple repos (bootloader, wireless, remote) and write wikis only about related repos in those repos (still need to accomodate all info on one page (can be done by [Github Pages](https://guides.github.com/features/pages/))).

Probably 2. is what You want.

**Comment by casainho** [comment]
Editing github wikis is fast for any user BUT that means I will not have track of changes (no notifications, no easy way to know who and what did changed). I do not want this. The other option to restrict to project collaborators only means it will be to much difficult because there is the need to do 2 PRs for 2 different repositories each change that includes adding / editing photos or other files.

I would like to keep the simplicity of MD files mixed with other documentation, on the same repo, so a download of the repo means the full project (not including the binaries releases). And I do not expect big editions from no members. PRs are more difficult than quick editing a wiki and I think that is favorable because it works as a filter, as I am not looking for anyone to edit. I plan to write all the documentation with the developers and not any user.

And [Github Pages](https://guides.github.com/features/pages/) seems to recommend this simplicity, were the page source is all in MD files and not on the wiki. And I would like to have a github page, because we could have a good simple looking page and simultaneous the technical repo page we are used to right now.

**Comment by stsdc** [comment]
> Editing github wikis is fast for any user BUT that means I will not have track of changes (no notifications, no easy way to know who and what did changed). I do not want this. The other option to restrict to project collaborators only means it will be to much difficult because there is the need to do 2 PRs for 2 different repositories each change that includes adding / editing photos or other files.

No, only one PR to ONE repository. You store images and things in *.wiki.git and not in regular one.

> And Github Pages seems to recommend this simplicity, were the page source is all in MD files and not on the wiki. And I would like to have a github page, because we could have a good simple looking page and simultaneous the technical repo page we are used to right now.

I can see. But we can also make a sync repository and everything will be inside Wiki.

**Comment by casainho** [comment]
> > Editing github wikis is fast for any user BUT that means I will not have track of changes (no notifications, no easy way to know who and what did changed). I do not want this. The other option to restrict to project collaborators only means it will be to much difficult because there is the need to do 2 PRs for 2 different repositories each change that includes adding / editing photos or other files.
> 
> No, only one PR to ONE repository. You store images and things in *.wiki.git and not in regular one.

Well, did not know that. Because that is not what is happening on TSDZ2_wiki... anyway, maybe the future is to write again the TSDZ2_wiki. At least I hope I will put my focus on the wireless version and not anymore on the displays, because displays are harder to develop for and I think that anyone wanting a good display should buy from the companies that do them with the ANT+ LEV.


> > And Github Pages seems to recommend this simplicity, were the page source is all in MD files and not on the wiki. And I would like to have a github page, because we could have a good simple looking page and simultaneous the technical repo page we are used to right now.
> 
> I can see. But we can also make a sync repository and everything will be inside Wiki.

Sorry, but for now it is clear to me that I would like to move forward with regular MD files AND Github pages. I was not remembering about the Github pages, you did talk about it and I think is a very good improvement to the project, to have less geeky, a more regular page.

Here an example of project. Repository for the documentation that is the same for the github page: https://github.com/HiveTracker/HiveTracker.github.io

image

And the resulting Github page:

image

**Comment by stsdc** [comment]
OK. So how it will look? Every repository has its own documentation and we will develop Github Page to accommodate all of them?

**Comment by casainho** [comment]
I think the project needs a main page for users, the ones that will install the wireless board, install the various firmware, the mobile app. That should be a not to much technical information (the one for developers). It should be as visual as possible with pictures or videos, when possible.

For developers, I think the documentation can be mainly on each repo. Also, the main page can have a small section with a link for other pages or the the pages on the other repos, specifically for developers.

**Comment by casainho** [comment]
@rananna and everyone,

I just created the https://opensourceebike.github.io as the main page for the project, that SHOULD only have documentation for final user (no techie) and not for developers. See previous discussion.

The documentation should have a short and good presentation of the EBike wireless (only for TSDZ2 currently) with the most user desirable features. Then, a only one easy, simple step by step process for installation and the same for usage.

Edit the main page here: https://github.com/OpenSourceEBike/OpenSourceEBike.github.io

I had to rename the organization to OpenSourceEBike, I removed the word firmware.

**Comment by rananna** [comment]
@casainho,
I certainly understand the wish to create a simplified presentation for non-technical users.
This type of thinking was behind my disappointment over not being able to update the signed bootloader using a simplified USB programming process that even non-technical people could easily understand. 
 I was forced to create a ' getting started' link for the remote control that involves new users navigating complicated programming steps that I am sure they would find daunting. 
However, I don't see how I can simplify these instructions any further.
Also, for this project new users also have to purchase boards, transistors, stlink programmers, switches, PCBs etc. and understand how to solder them correctly. This is hardly a geek free experience.
If some company offers to sell a TSDZ2 motor with our wireless capability in the future that would eliminate all the complexity for new users, but for now, it is a DIY project and complexity is very hard to avoid.
Maybe we need to revisit getting boards made by a third party like https://jlcpcb.com/smt-assembly to eliminate the barrier to entry.

Anyway, keeping these types of restrictions in mind, it is certainly possible to create a simplified github page view for new users that eliminates much of additional information provided for developers.

Although I could make an attempt at editing the main page and add information for the remote control,  I think I would like to understand more of what you want to see on this page as your personal style preference  has a big part to play in this . 
You probably have a very clear idea of what you're looking for and I could be wasting my time developing a page view that is not compatible with your vision.
I would like to suggest that you modify index.md to create a style template for me to follow for the new user presentation you have in mind.

**Comment by casainho** [comment]
> Also, for this project new users also have to purchase boards, transistors, stlink programmers, switches, PCBs etc. and understand how to solder them correctly. This is hardly a geek free experience.

Yes, but there are differences from the ones that can solder a simple wires on Arduino and others that work professionally on electronics. But seems we are on the same page about this.

> If some company offers to sell a TSDZ2 motor with our wireless capability in the future that would eliminate all the complexity for new users, but for now, it is a DIY project and complexity is very hard to avoid.
> Maybe we need to revisit getting boards made by a third party like https://jlcpcb.com/smt-assembly to eliminate the barrier to entry.

Maybe the shops will want to invest on doing a production quality boards, for now I got not contact of interest. I guess will be easier for everyone to understand all this when it is implemented and are other users doing video reviews. I must say this is a bit complex, the wireless, and not everyone has experience with Garmins to understand all this. And there is also some risk some parts of all this will not work well for final users, may be a bit complex and a display that always works will be simple for the average user.

> Although I could make an attempt at editing the main page and add information for the remote control, I think I would like to understand more of what you want to see on this page as your personal style preference has a big part to play in this .
> You probably have a very clear idea of what you're looking for and I could be wasting my time developing a page view that is not compatible with your vision.
> I would like to suggest that you modify index.md to create a style template for me to follow for the new user presentation you have in mind.

I will but first I need to install on my ebike and then I will understand the needed steps and being able to document it.

**Pull request merged**